tests/kafkatest/tests/connect/connect_plugins_rest_test.py

```diff
@@ -0,0 +1,250 @@
+    }
+
+    FILE_SINK_CONFIGS = {
+        'file': {'required': True, 'type': 'STRING', 'importance': 'HIGH', 'group': None, 'order': -1, 'width': 'NONE'}
+    }
+
+    FILE_SOURCE_ALL_CONFIGS = merge_two_dicts(COMMON_CONNECTOR_CONFIGS, FILE_SOURCE_CONFIGS)
+    FILE_SINK_ALL_CONFIGS = merge_two_dicts(COMMON_CONNECTOR_CONFIGS, FILE_SINK_CONFIGS)
+
+    FILE_SOURCE_CONNECTOR = 'org.apache.kafka.connect.file.FileStreamSourceConnector'
+    FILE_SOURCE_CONNECTOR_SHORT = 'FileStreamSourceConnector'
+    FILE_SOURCE_CONNECTOR_ALIAS = 'FileStreamSource'
+
+    FILE_SINK_CONNECTOR = 'org.apache.kafka.connect.file.FileStreamSinkConnector'
+    FILE_SINK_CONNECTOR_SHORT = 'FileStreamSinkConnector'
+    FILE_SINK_CONNECTOR_ALIAS = 'FileStreamSink'
+
+    def __init__(self, test_context):
+        super(ConnectPluginsRestApiTest, self).__init__(test_context, num_zk=1, num_brokers=1)
+        self.cc = ConnectDistributedService(test_context, 2, self.kafka)
+
+    def test_rest_api(self):
+        self.cc.set_configs(lambda node: self.render("connect-distributed.properties", node=node))
+
+        self.cc.start()
+
+        expected = {self.FILE_SOURCE_CONNECTOR, self.FILE_SINK_CONNECTOR}
+        connector_plugins = self.cc.list_connector_plugins()
+        connector_plugins_set = set([connector_plugin['class'] for connector_plugin in connector_plugins])
+        self.validate_plugins(expected, connector_plugins_set)
+        self.validate_config_def(self.FILE_SOURCE_CONNECTOR, self.COMMON_CONNECTOR_CONFIGS, {})
+        self.validate_config_def(self.FILE_SOURCE_CONNECTOR_SHORT, self.COMMON_CONNECTOR_CONFIGS, {})
+        self.validate_config_def(self.FILE_SOURCE_CONNECTOR_ALIAS, self.COMMON_CONNECTOR_CONFIGS, {})
+
+        self.validate_config_def(self.FILE_SINK_CONNECTOR, self.COMMON_CONNECTOR_CONFIGS, {})
+        self.validate_config_def(self.FILE_SINK_CONNECTOR_SHORT, self.COMMON_CONNECTOR_CONFIGS, {})
+        self.validate_config_def(self.FILE_SINK_CONNECTOR_ALIAS, self.COMMON_CONNECTOR_CONFIGS, {})
+
+        # Test the case that we have invalid values in connector common configs
+        props = {
+            'name': 'test',
+            'connector.class': self.FILE_SOURCE_CONNECTOR,
+            'tasks.max': 0
+        }
+
+        expected_config_values = {
+            'name': {'name': 'name', 'value': 'test', 'errors': [], 'recommended_values': [], 'visible': True},
+            'connector.class': {'name': 'connector.class', 'value': self.FILE_SOURCE_CONNECTOR, 'errors': [], 'recommended_values': [], 'visible': True},
+            'tasks.max': {'name': 'tasks.max', 'value': 0, 'errors': ['Invalid value 0 for configuration tasks.max: Value must be at least 1'], 'recommended_values': [], 'visible': True},
+            'topics': {'name': 'topics', 'value': [], 'errors': [], 'recommended_values': [], 'visible': True}
+        }
+
+        configs = self.validate_config_def(self.FILE_SOURCE_CONNECTOR, self.FILE_SOURCE_ALL_CONFIGS, props)
```

Comment:
This tests quite a bit which is largely redundant given the unit tests, right? System tests are more useful to make sure everything works when integrated together. This tests some of that (e.g. that you actually can validate against a real connector plugin), but a bunch of it seems to be validating details that could be confirmed via unit tests (e.g. validating in detail the output format). Would something much smaller and integrated with the existing `connect_rest_test.py` be enough here? That would also avoid adding a whole additional test (which has non-trivial setup/teardown overhead).